tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,739 @@
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-darwin-x64.zip</Key>
+              <LastModified>2017-01-17T01:18:56.000Z</LastModified>
+              <ETag>"81b438a9f7b2d4c871cfbd5aedd96975"</ETag>
+              <Size>139834277</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-linux-x64.zip</Key>
+              <LastModified>2017-01-17T02:01:01.000Z</LastModified>
+              <ETag>"35660b65233082a10c00828ea1e50c38"</ETag>
+              <Size>55960697</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-linux-x86.zip</Key>
+              <LastModified>2017-01-17T02:18:20.000Z</LastModified>
+              <ETag>"1bafcc0d5d2c8d43bd2ce8948bb51a8b"</ETag>
+              <Size>57331229</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x64.exe</Key>
+              <LastModified>2017-01-17T05:48:02.000Z</LastModified>
+              <ETag>"b0a6154ec79d17618632ac62eb30b44e"</ETag>
+              <Size>84802632</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x64.zip</Key>
+              <LastModified>2017-01-17T06:14:46.000Z</LastModified>
+              <ETag>"f25c5dfa8378e608c25fafbe65e90162"</ETag>
+              <Size>82235087</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x86.exe</Key>
+              <LastModified>2017-01-17T06:42:58.000Z</LastModified>
+              <ETag>"183b6eb648b0a78a1e99c9182f90194e"</ETag>
+              <Size>74264232</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x86.zip</Key>
+              <LastModified>2017-01-17T07:05:55.000Z</LastModified>
+              <ETag>"d8d860013f038bed3f52534053b08382"</ETag>
+              <Size>72042525</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+          </ListBucketResult>
+        `);
+      });
+
+      afterEach(function() {
+        nock.cleanAll();
+      });
+
+      it('should not consider the other packages', function(done) {
+        s3Packages.getRemoteVersions(s3Packages.BUCKET_URL.PRODUCTION).then((versions) => {
+          m.chai.expect(versions).to.deep.equal([
+            '1.0.0-beta.17'
+          ]);
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given an unsuccessful request', function() {
```

Comment:
Should we also test a successful request that doesn't return _any_ versions of Etcher?
(e.g. an empty bucket, or a bucket only containing non-Etcher packages)